test/extended/images/sample_repos.go

```diff
@@ -44,6 +44,14 @@ func NewSampleRepoTest(c SampleRepoConfig) func() {
 				err := oc.Run("new-app").Args("-f", c.templateURL).Execute()
 				o.Expect(err).NotTo(o.HaveOccurred())
 
+				// sometimes the deployment controller takes a while to start the DB deployment which
+				// results in the app coming up before the DB is available, which makes the app tests fail
+				// because they require the DB is available when the app started, not when the http request
+				// is made (they init the DB when they start, not on each request).  So here we force
+				// an immediate deployment of the DB.  It may error because there is already a deployment
+				// in progress, so do not check for an error.
+				oc.Run("deploy").Args(c.dbDeploymentConfigName, "--latest").Execute()
```

Comment:
I see.  OK, yeah, that makes what has been occurring explainable.  And
certainly provides some potential things in the future to look into from
the app side of things if we see varying behavior again.

lgtm

On Sun, Jun 26, 2016 at 7:40 PM, Ben Parees <notifications@github.com>
wrote:

> In test/extended/images/sample_repos.go
> <https://github.com/openshift/origin/pull/9567#discussion_r68512429>:
>
> > @@ -44,6 +44,14 @@ func NewSampleRepoTest(c SampleRepoConfig) func() {
> >  				err := oc.Run("new-app").Args("-f", c.templateURL).Execute()
> >  				o.Expect(err).NotTo(o.HaveOccurred())
> >
> > +				// sometimes the deployment controller takes a while to start the DB deployment which
> > +				// results in the app coming up before the DB is available, which makes the app tests fail
> > +				// because they require the DB is available when the app started, not when the http request
> > +				// is made (they init the DB when they start, not on each request).  So here we force
> > +				// an immediate deployment of the DB.  It may error because there is already a deployment
> > +				// in progress, so do not check for an error.
> > +				oc.Run("deploy").Args(c.dbDeploymentConfigName, "--latest").Execute()
>
> I think the precise explanation is that perhaps only the nodejs sample app
> initializes the DB during startup and never again attempts to get a DB
> connection. I haven't reviewed all the other apps, but i think at least
> some of them either retry until they get a DB connection before coming up
> at all, or they will re-attempt to get a DB connection on each request, if
> they do not have one.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9567/files/3799a55c75d873ab1c5f9960d288a7768903e7ee#r68512429>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADbadKO9DTUeXw-jgbr1GDATofZ8LtsQks5qPw4BgaJpZM4I-elp>
> .
>